### ISSUE TITLE
OSError: No space left on device. Changing working directory not solving the issue. 

### ISSUE THREAD

**Issue by bwinsto2** [poldracklab/fmriprep#2120]
Running fMRIprep 20.0.6 with fmriprep-docker on linux. fMRIprep was running great until I ran out of space on _something_. I tried running the pipeline again but specifying the working directory this time as suggested in #1291, but no dice: I got the same error.
Here is the code for when I ran it the second time (with the working directory somewhere with a lot of space): 
```
fmriprep-docker /data2/me/.../BIDS/raw_data /data2/me/.../fMRIprep/output_May_7 participant --participant-label jrd722 
--fs-license-file /home/me/.local/bin/license.txt -w /data2/me/.../fMRIprep/temp
RUNNING: docker run --rm -it -e DOCKER_VERSION_8395080871=19.03.8 -v /home/me/.local/bin/license.txt:/opt/freesurfer/license.txt:ro 
-v /data2/.../BIDS/raw_data:/data:ro -v /data2/.../fMRIprep/output_May_7:/out 
-v /data2/.../fMRIprep/temp:/scratch poldracklab/fmriprep:20.0.6 /data /out participant --participant-label jrd722 -w /scratch
```
And the error on the very first process:
```
OSError: [Errno 28] No space left on device: '/home/fmriprep/.cache/matplotlib/tex.cache'
```
Of course, the error I got on the first run was for a different directory (I didn't specify a working directory on this first run):
```
OSError: [Errno 28] No space left on device: '/tmp/work/fmriprep_wf/single_subject_jrd722_wf/func_preproc_ses_Session1_task_med_run_3_wf/bold_std_trans_wf/bold_reference_wf/_std_target_MNI152NLin2009cAsym./gen_ref/result_gen_ref.pklz.tmp'
```
I'm just not sure what exactly is running out of space, or how I can figure it out. 

I will note that there is a directory that has gotten suspiciously large: `/var/lib/docker/overlay2` has grown to 83GB since running the pipeline a couple times. I actually ran into this "no space left on device" issue last week, and I figured this docker/overlay2 was to blame since it was 144GB at the time. To test it out, I deleted the contents of overlay2 (I know, I know, bad idea), which required me to reinstall docker. However, when I ran fMRIprep again (with overlay2 relatively empty), it worked! AKA no out of space error. This leads me to believe overlay2 is part of the problem; I'm not sure why, for example, it was 144GB last time and only 83GB this time before running out of space. I'm not even sure if it's part of the problem, but I thought I would mention it. Obviously, I don't want to simply delete everything in the directory again because that's not a real way of fixing this problem, and as I've noted, it's a very temporary solution. 

Any help in resolving this would be very much appreciated. Thank you.

**Comment by effigies** [comment]
The device that's running out of space is the one hosting your Docker containers.

Use `docker images` to see all images downloaded, and `docker ps -a` to see all containers either running or finished but not deleted. Images can be removed with `docker rmi` and containers with `docker rm`.

**Comment by bwinsto2** [comment]
Thanks @effigies. I cleared out every docker container I had and reran the pipeline. This time, it went much, much further but still ran out of space (the size of /var/lib/docker went from hardly anything to 90GB--is this normal?). I suppose I will have to look into expanding the allowable space for the docker directory. Thanks again.

**Comment by effigies** [comment]
The only place we should be writing into the container by default is `/tmp`, and that shouldn't happen if you pass `-w`. If you can open up the container `docker exec <container> bash` (I think... you might want to Google how to open a shell in a running container if that doesn't work), I would run `du -sh /*` to try to track down where 90GB is being stored.

**Comment by bwinsto2** [comment]
Ah--this latest time I ran it, I had the working directory writing inside the container again since I thought there might be enough space after I cleared out the other containers. I will clear everything out again and specify the wd on my machine. That should solve it--will update. Thanks!!

**Comment by bwinsto2** [comment]
Got it working! I think part of my problem with setting the working directory was that previously I had created a /temp folder on my machine prior to running the pipeline (and it never populated).
When I deleted that folder and let the pipeline create its own temp and output folders it no longer was writing the files inside the docker container (which led to the no space left on device issue). 
Thanks again for the help!
